### PR TITLE
GEODE-6014: Removed unnecessary null checks.

### PR DESCRIPTION
**Pull request by nabarunnag**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/distributed/internal/ClusterDistributionManager.java [hunk 1]
```diff
@@ -1314,7 +1314,7 @@ public void addHostedLocators(InternalDistributedMember member, Collection<Strin
       this.hostedLocatorsAll = tmp;
 
       if (isSharedConfigurationEnabled) {
-        if (locators == null || locators.isEmpty()) {
+        if (locators.isEmpty()) {
```
Comment (bschuchardt) [inline geode-core/src/main/java/org/apache/geode/distributed/internal/ClusterDistributionManager.java]: I think you can remove this if statement.  We've already checked if locators.isEmpty() earlier in the method.

File: geode-core/src/main/java/org/apache/geode/internal/concurrent/CompactConcurrentHashSet2.java [hunk 2]
```diff
@@ -353,8 +353,8 @@ public String toString() {
     public boolean equals(Object o) {
       Object k, u;
```
Comment (bschuchardt) [inline geode-core/src/main/java/org/apache/geode/internal/concurrent/CompactConcurrentHashSet2.java]: It looks like you can delete these variables

File: geode-core/src/main/java/org/apache/geode/internal/concurrent/CompactConcurrentHashSet2.java [hunk 3]
```diff
@@ -1711,14 +1711,13 @@ boolean removeTreeNode(TreeNode<K> p) {
               else
                 sp.right = p;
             }
-            if ((s.right = pr) != null)
-              pr.parent = s;
+
```
Comment (bschuchardt) [inline geode-core/src/main/java/org/apache/geode/internal/concurrent/CompactConcurrentHashSet2.java]: Why is it okay to delete the assignment (s.right = pr)?

File: geode-core/src/main/java/org/apache/geode/internal/concurrent/CompactConcurrentHashSet2.java [hunk 4]
```diff
@@ -1711,14 +1711,13 @@ boolean removeTreeNode(TreeNode<K> p) {
...
+            pr.parent = s;
           }
           p.left = null;
           if ((p.right = sr) != null)
             sr.parent = p;
-          if ((s.left = pl) != null)
-            pl.parent = s;
+          pl.parent = s;
```
Comment (bschuchardt) [inline geode-core/src/main/java/org/apache/geode/internal/concurrent/CompactConcurrentHashSet2.java]: Why is it okay to delete the assignment (s.left = pl)?

File: geode-core/src/main/java/org/apache/geode/internal/concurrent/CompactConcurrentHashSet2.java (hunk 2 above)
Comment (nabarunnag) [reply geode-core/src/main/java/org/apache/geode/internal/concurrent/CompactConcurrentHashSet2.java]: done

File: geode-core/src/main/java/org/apache/geode/internal/concurrent/CompactConcurrentHashSet2.java (hunk 4 above)
Comment (nabarunnag) [reply geode-core/src/main/java/org/apache/geode/internal/concurrent/CompactConcurrentHashSet2.java]: I made a mistake... 
Used the intelliJ auto-correct.
--fixed

File: geode-core/src/main/java/org/apache/geode/internal/cache/FilterProfile.java [hunk 5]
```diff
@@ -1170,10 +1170,10 @@ public void getLocalFilterRoutingForPutAllOp(DistributedPutAllOperation dpao,
       int size = putAllData.length;
       CqService cqService = getCqService(dpao.getRegion());
       boolean doCQs = cqService.isRunning()
-          && this.region != null /*
-                                  * && !(this.region.isUsedForPartitionedRegionBucket() ||
-                                  * (this.region instanceof PartitionedRegion))
-                                  */;
+      /*
```
Comment (mcmellawatt) [inline geode-core/src/main/java/org/apache/geode/internal/cache/FilterProfile.java]: Can we just delete this commented out code?

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/util/ReadWriteFile.java [hunk 6]
```diff
@@ -150,13 +150,13 @@ public static String readWriteFile(String logFileName, String logToBeWritten, St
...
-      if (input != null) {
-        input.close();
-      }
-      if (output != null) {
-        output.flush();
-        output.close();
-      }
+
+      input.close();
+
```
Comment (mcmellawatt) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/util/ReadWriteFile.java]: Extra newline here maybe?

**Comment by nabarunnag** [comment]
@mcmellawatt These were initially flagged by LGTM, but they were reviewed in the code. If you load the diff using an IDE you can see the reason why these values can't be null

File: geode-core/src/main/java/org/apache/geode/internal/cache/FilterProfile.java (hunk 5 above)
Comment (nabarunnag) [reply geode-core/src/main/java/org/apache/geode/internal/cache/FilterProfile.java]: fixed

**Pull request merged**